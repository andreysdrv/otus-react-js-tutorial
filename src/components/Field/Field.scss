.field {
  background-color: #f9ffb5;
  display: flex;
  flex-wrap: wrap;
}