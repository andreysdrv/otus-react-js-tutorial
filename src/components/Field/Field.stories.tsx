import React from "react";
import Field from "./Field";
import { ComponentMeta, ComponentStory } from "@storybook/react";
import { CELLS, FIELD_SIZE } from "../utils/config";

export default {
  component: Field,
  title: "Field",
  decorators: [(story) => <div style={{ padding: "1rem" }}>{story()}</div>],
} as ComponentMeta<typeof Field>;

const Template: ComponentStory<typeof Field> = (args) => <Field {...args} />;

export const Default = Template.bind({});
Default.args = {
  size: FIELD_SIZE,
  values: CELLS,
};
