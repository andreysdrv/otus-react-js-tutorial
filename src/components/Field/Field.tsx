import React, { FC } from "react";
import "./Field.scss";
import Cell from "../Cell";

interface FieldProps {
  size: number;
  values: number[];
}

const Field: FC<FieldProps> = ({ size, values }) => {
  const handleCellClick = (cell) => console.log(cell);

  return (
    <div
      className="field"
      style={{ width: `${size * 100}px`, height: `${size * 100}px` }}
    >
      {values.map((cell) => (
        <Cell key={cell} content={cell} handleCellClick={handleCellClick} />
      ))}
    </div>
  );
};

export default Field;
