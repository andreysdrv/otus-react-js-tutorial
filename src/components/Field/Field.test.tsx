import React from "react";
import { render } from "@testing-library/react";
import Field from "./Field";

describe("Field snapshot tests", () => {
  it("base snapshot", () => {
    const { container } = render(<Field size={2} values={[1, 2, 3, 4]} />);
    expect(container).toMatchSnapshot();
  });

  it("some data snapshot", () => {
    const { container } = render(<Field size={2} values={[1, 2, 3, 4]} />);
    expect(container).toMatchSnapshot();
  });
});
