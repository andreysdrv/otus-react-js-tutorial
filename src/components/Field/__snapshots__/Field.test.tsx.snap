// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Field snapshot tests base snapshot 1`] = `
<div>
  <div
    class="field"
    style="width: 200px; height: 200px;"
  >
    <div
      class="cell"
    >
      1
    </div>
    <div
      class="cell"
    >
      2
    </div>
    <div
      class="cell"
    >
      3
    </div>
    <div
      class="cell"
    >
      4
    </div>
  </div>
</div>
`;

exports[`Field snapshot tests some data snapshot 1`] = `
<div>
  <div
    class="field"
    style="width: 200px; height: 200px;"
  >
    <div
      class="cell"
    >
      1
    </div>
    <div
      class="cell"
    >
      2
    </div>
    <div
      class="cell"
    >
      3
    </div>
    <div
      class="cell"
    >
      4
    </div>
  </div>
</div>
`;
